sdk/core/azure-core/test/ut/curl_options_test.cpp

```diff
@@ -234,6 +235,53 @@ namespace Azure { namespace Core { namespace Test {
                         .ConnectionPoolIndex.clear());
   }
 
+#if defined(SUPPORT_SETTING_CAPATH)
+  TEST(CurlTransportOptions, setCADirectory)
+  {
+    Azure::Core::Http::CurlTransportOptions curlOptions;
+    // openssl default cert location will be used only if environment variable SSL_CERT_DIR
+    // is not set
+    const char* ca = getenv(X509_get_default_cert_dir_env());
+    if (ca)
+    {
+      curlOptions.CAPath = ca;
+    }
+    else
+    {
+      curlOptions.CAPath = X509_get_default_cert_dir();
+    }
+
+    auto transportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>(curlOptions);
+    Azure::Core::Http::Policies::TransportOptions options;
+    options.Transport = transportAdapter;
+    auto transportPolicy
+        = std::make_unique<Azure::Core::Http::Policies::_internal::TransportPolicy>(options);
+
+    std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> policies;
+    policies.emplace_back(std::move(transportPolicy));
+    Azure::Core::Http::_internal::HttpPipeline pipeline(policies);
+
+    // Use HTTPS
+    Azure::Core::Url url(AzureSdkHttpbinServer::Get());
+    Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Get, url);
+
+    std::unique_ptr<Azure::Core::Http::RawResponse> response;
+    EXPECT_NO_THROW(response = pipeline.Send(request, Azure::Core::Context::ApplicationContext));
+    auto responseCode = response->GetStatusCode();
+    int expectedCode = 200;
+    EXPECT_PRED2(
+        [](int expected, int code) { return expected == code; },
+        expectedCode,
+        static_cast<typename std::underlying_type<Azure::Core::Http::HttpStatusCode>::type>(
+            responseCode));
```

Comment:
Am I missing something? Why not this? -
```suggestion
    EXPECT_EQ(response->GetStatusCode(), Azure::Core::Http::HttpStatusCode::Ok);
```
```suggestion
    auto responseCode = response->GetStatusCode();
    int expectedCode = 200;
    EXPECT_PRED2(
        [](int expected, int code) { return expected == code; },
        expectedCode,
        static_cast<typename std::underlying_type<Azure::Core::Http::HttpStatusCode>::type>(
            responseCode));
```

Reply:
I totally agree with the changes. I was trying to align with the `sslVerifyOff`test above but I was not sure the purpose of it either atm as well.